### PR TITLE
[libc][math][c23] Add cosf16 function

### PR DESCRIPTION
**Pull request by wldfngrs**
- Implementation of cos for 16-bit floating point inputs
- Exhaustive tests across the 16-bit input range

### CODE REVIEW

**Review by overmighty — COMMENTED** [review]
Sorry for the delay. Could you please update the PR and resolve the merge conflict?

File: libc/src/math/generic/cosf16.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  // formula:
+  //   cos(x) = cos((k + y) * pi/32)
+  //          = cos(k * pi/32) * cos(y * pi/32) -
+  //            sin(k * pi/32) * sin(y * pi/32)
+
+  // Handle exceptional values
+
+  if (LIBC_UNLIKELY(x_abs == 0x2b7c || x_abs == 0x4ac1 || x_abs == 0x5c49 ||
+                    x_abs == 0x7acc)) {
+    if (auto r = COSF16_EXCEPTS.lookup(x_abs); LIBC_UNLIKELY(r.has_value()))
```
[the remark this answers is not in the document]
Comment (wldfngrs) [reply libc/src/math/generic/cosf16.cpp:60]: smh, good pick, it *is* redundant work. I'd fix that up, immediately

**Review by overmighty — APPROVED** [review]

**Comment by overmighty** [comment]
Do you want me to merge this for you?

**Comment by wldfngrs** [comment]
> Do you want me to merge this for you?

Yep, that's fine.

**Pull request merged**